src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -87,6 +87,8 @@
  */
 class ProductController extends FrameworkBundleAdminController
 {
+    public const FEATURE_FLAG_PRODUCT_PAGE_V2 = 'product_page_v2';
```

Comment:
I'm not sure this const should be in this controller, nor in the new one Because they become co-dependent just to get a feature flag name You could simply use a hardcoded value since it's the ID/slug/name a dynamic data in the database (a bit like we do with Configuration keys)

If you really want to use a const as a reference (which I agree is cleaner) it should be in an independent object that is related to FeatureFlag, not the controller This will avoid co-dependency and besides the `product v2` feature is not only limited to one controller (it's composed of several ones, plus CQRS command, plus the list, ...)

So I'd suggest creating either a generic `FeatureFlagSettings` class to hold such const, either a generic one which would include future other flags along time or a specific one `ProductFeatureFlagSettings` which is dedicated to product

Reply:
Fixed https://github.com/PrestaShop/PrestaShop/pull/24038/commits/feb0b3807ead2c1f0757aadae6a4c3ca5fc4988f